dao/statistics.go

```diff
@@ -100,6 +101,13 @@ func (d statisticsDao) GetCourseStatsHourly(courseID uint) ([]Stat, error) {
 	return res, err
 }
 
+// GetCurrentLiveViewers returns the number of viewers currently watching a live stream.
+func (d statisticsDao) GetCurrentLiveViewers(streamID uint) (int, error) {
+	var res int
+	err := DB.Raw(`SELECT viewers FROM stats WHERE stream_id = ? AND live = 1`, streamID).Scan(&res).Error
```

Comment:
This should give you only the latest stat entry:

```suggestion
	err := DB.Raw(`SELECT viewers FROM stats WHERE stream_id = ? AND live = 1 ORDER BY id DESC LIMIT 1`, streamID).Scan(&res).Error
```

Reply:
Thanks, nice catch!